marshal_10_decimal_test.go

```diff
@@ -0,0 +1,52 @@
+package gocql
+
+import (
+	"github.com/gocql/gocql/internal/tests/utils"
+	"gopkg.in/inf.v0"
+	"math"
+	"testing"
+
+	"github.com/gocql/gocql/marshal/tests/mod"
+	"github.com/gocql/gocql/marshal/tests/serialization"
+)
+
+func TestMarshalDecimal(t *testing.T) {
+	marshal := func(i interface{}) ([]byte, error) { return Marshal(NativeType{proto: 4, typ: TypeDecimal}, i) }
+	unmarshal := func(bytes []byte, i interface{}) error {
+		return Unmarshal(NativeType{proto: 4, typ: TypeDecimal}, bytes, i)
+	}
+
+	brokenTypes := utils.GetTypes(inf.Dec{}, (*inf.Dec)(nil))
```

Comment:
Can I ask you to add a comment that breafly describe why these types are broken

Reply:
Done